src/libs/nav2_util/CMakeLists.txt

```diff
@@ -17,18 +17,43 @@ find_package(tf2 REQUIRED)
 
 include_directories(include)
 
-add_library(nav2_util STATIC
+add_library(costmap_lib SHARED
   src/costmap.cpp
 )
 
-ament_target_dependencies(nav2_util
+ament_target_dependencies(costmap_lib
   nav2_libs_msgs
   tf2_ros
-  tf2
+  tf2)
+
+add_library(map_lib SHARED
+  src/map/map.c
+  src/map/map_store.c
+  src/map/map_range.c
+  src/map/map_draw.c
+  src/map/map_cspace.cpp
+)
+
+add_library(pf_lib SHARED
+  src/pf/pf.c
+  src/pf/pf_kdtree.c
+  src/pf/pf_pdf.c
+  src/pf/pf_vector.c
+  src/pf/eig3.c
+  src/pf/pf_draw.c
+)
+
+add_library(sensors_lib SHARED
```

Comment:
The sensor library has some dependencies on ROS messages types. There should be an ament_target_dependencies for sensors_lib. Please file an issue on this (there can be one issue that addresses this and the previous comment.

Reply:
Will do.  Thanks.

Reply:
No it doesn't have dependencies on ROS message types.  All the poses and odom messages are defined internally within the sensor and pf library.